dali/kernels/imgproc/warp/mapping_traits.h

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_WARP_MAPPING_TRAITS_H_
+#define DALI_KERNELS_IMGPROC_WARP_MAPPING_TRAITS_H_
+
+#include <functional>
+#include "dali/core/geom/vec.h"
+
+namespace dali {
+namespace kernels {
+namespace warp {
+
+std::true_type is_fp_mapping_helper(const std::function<vec2(vec2)> &f);
+std::true_type is_fp_mapping_helper(const std::function<vec3(vec3)> &f);
+std::true_type is_fp_mapping_helper(const std::function<vec4(vec4)> &f);
+
+std::false_type is_fp_mapping_helper(const std::function<ivec2(ivec2)> &f);
+std::false_type is_fp_mapping_helper(const std::function<ivec3(ivec3)> &f);
+std::false_type is_fp_mapping_helper(const std::function<ivec4(ivec4)> &f);
+
+
+template <typename Mapping>
+struct is_fp_mapping : decltype(is_fp_mapping_helper(Mapping())) {};
+
+template <typename Mapping>
+struct mapping_params {
+  using type = Mapping;
+};
+
+template <typename Mapping>
+using mapping_params_t = typename mapping_params<Mapping>::type;
```

Comment:
Could you provide short comment, what is this trait for?

Reply:
I can. I can also explain it now:
The `Mapping` type is something that actually _does_ the coordinate mapping.
`mapping_params_t`, on the other hand, is an object that contains information necessary and sufficient to create a functional Mapping object. For example, the jitter Mapping will most likely contain some random number generator (or, perhaps, many of them), but the mapping parameters will be sampling coordinate distributions.